Docker host access flag causing panic

## Describe the bug
The new `--allow-docker-host-access` flag causes a panic

## To Reproduce
Steps to reproduce the behavior:
1. Run `porter install` using the `--allow-docker-host-access` flag

## Expected behavior
No panic :)

## Porter Command and Output
```
$ porter install  --debug --allow-docker-host-access
...
Resolved storage plugin to storage.porter.filesystem
<my_homedir>/.porter/porter plugin run storage.porter.filesystem
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x229f483]

goroutine 1 [running]:
get.porter.sh/porter/pkg/cnab/provider.(*Runtime).dockerDriverWithHostAccess.func1(0xc0002fe8c0, 0xc00056ed80, 0x10, 0x10)
	/home/vsts/work/1/s/gopath/src/get.porter.sh/porter/pkg/cnab/provider/driver.go:78 +0x33
github.com/cnabio/cnab-go/driver/docker.(*Driver).exec(0xc000402540, 0xc00042cd00, 0x0, 0x0, 0x0)
	/home/vsts/work/1/s/gopath/pkg/mod/github.com/cnabio/cnab-go@v0.10.1-beta1/driver/docker/docker.go:168 +0x4ed
github.com/cnabio/cnab-go/driver/docker.(*Driver).Run(0xc000402540, 0xc00042cd00, 0x3, 0xc00042cd00, 0x0)
	/home/vsts/work/1/s/gopath/pkg/mod/github.com/cnabio/cnab-go@v0.10.1-beta1/driver/docker/docker.go:38 +0x35
github.com/cnabio/cnab-go/action.(*Install).Run(0xc0006395b8, 0xc000301c70, 0xc000551620, 0xc0003c64e0, 0x3, 0x3, 0xc0003d18c0, 0xc0004549a0)
	/home/vsts/work/1/s/gopath/pkg/mod/github.com/cnabio/cnab-go@v0.10.1-beta1/action/install.go:31 +0x176
get.porter.sh/porter/pkg/cnab/provider.(*Runtime).Install(0xc00049b890, 0xc0002265f0, 0x9, 0xc0002a0c40, 0x11, 0xc0003d18f0, 0xc0003d18c0, 0xc0004549a0, 0x1, 0x1, ...)
	/home/vsts/work/1/s/gopath/src/get.porter.sh/porter/pkg/cnab/provider/install.go:73 +0x8e6
get.porter.sh/porter/pkg/porter.(*Porter).InstallBundle(0xc0001ad400, 0x27f97ab, 0xb, 0xc0002a0c40, 0x11, 0x0, 0x0, 0xc0002265f0, 0x9, 0xc00019e800, ...)
	/home/vsts/work/1/s/gopath/src/get.porter.sh/porter/pkg/porter/install.go:47 +0x3cf
main.buildBundleInstallCommand.func2(0xc00020b180, 0xc000207a00, 0x0, 0x10, 0x0, 0x0)
	/home/vsts/work/1/s/gopath/src/get.porter.sh/porter/cmd/porter/bundle.go:114 +0x7e
github.com/spf13/cobra.(*Command).execute(0xc00020b180, 0xc000207700, 0x10, 0x10, 0xc00020b180, 0xc000207700)
	/home/vsts/work/1/s/gopath/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002b2a00, 0x0, 0x24edbc0, 0xc00009c058)
	/home/vsts/work/1/s/gopath/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/home/vsts/work/1/s/gopath/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
	/home/vsts/work/1/s/gopath/src/get.porter.sh/porter/cmd/porter/main.go:18 +0x27
```

## Version
porter v0.26.0-beta.1 (b15c4109)

Tried using canary build, v0.26.0-beta.1-3-g1aa1a72f (1aa1a72f), and same error

--- a/pkg/cnab/extensions/docker.go
+++ b/pkg/cnab/extensions/docker.go
@@ -43,12 +43,25 @@ func DockerExtensionReader(bun *bundle.Bundle) (interface{}, error) {
 			DockerExtensionKey, string(dataB))
 	}
 
-	dha := &Docker{}
-	err = json.Unmarshal(dataB, dha)
+	dha := Docker{}
+	err = json.Unmarshal(dataB, &dha)
 	if err != nil {
 		return nil, errors.Wrapf(err, "could not unmarshal the %q extension %q",
 			DockerExtensionKey, string(dataB))
 	}
 
 	return dha, nil
+}
+
+// GetDockerExtension checks if the docker extension is present and returns its
+// extension configuration.
+func (e ProcessedExtensions) GetDockerExtension() (dockerExt Docker, dockerRequired bool, err error) {
+	ext, extensionRequired := e[DockerExtensionKey]
+
+	dockerExt, ok := ext.(Docker)
+	if !ok && extensionRequired {
+		return Docker{}, extensionRequired, errors.Errorf("unable to parse Docker extension config: %+v", dockerExt)
+	}
+
+	return dockerExt, extensionRequired, nil
 }

--- a/pkg/cnab/extensions/docker_test.go
+++ b/pkg/cnab/extensions/docker_test.go
@@ -1,0 +1,43 @@
+package extensions
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProcessedExtensions_GetDockerExtension(t *testing.T) {
+	t.Run("extension present", func(t *testing.T) {
+		ext := ProcessedExtensions{
+			DockerExtensionKey: Docker{
+				Privileged: true,
+			},
+		}
+
+		dockerExt, dockerRequired, err := ext.GetDockerExtension()
+		require.NoError(t, err, "GetDockerExtension failed")
+		assert.True(t, dockerRequired, "docker should be a required extension")
+		assert.Equal(t, Docker{Privileged: true}, dockerExt, "docker config was not populated properly")
+	})
+
+	t.Run("extension missing", func(t *testing.T) {
+		ext := ProcessedExtensions{}
+
+		dockerExt, dockerRequired, err := ext.GetDockerExtension()
+		require.NoError(t, err, "GetDockerExtension failed")
+		assert.False(t, dockerRequired, "docker should NOT be a required extension")
+		assert.Equal(t, Docker{}, dockerExt, "Docker config should default to empty when not required")
+	})
+
+	t.Run("extension invalid", func(t *testing.T) {
+		ext := ProcessedExtensions{
+			DockerExtensionKey: map[string]string{"ponies": "are great"},
+		}
+
+		dockerExt, dockerRequired, err := ext.GetDockerExtension()
+		require.Error(t, err, "GetDockerExtension failed")
+		assert.True(t, dockerRequired, "docker should be a required extension")
+		assert.Equal(t, Docker{}, dockerExt, "Docker config should default to empty")
+	})
+}

--- a/pkg/cnab/provider/driver.go
+++ b/pkg/cnab/provider/driver.go
@@ -25,20 +25,19 @@ func (r *Runtime) newDriver(driverName string, claimName string, args ActionArgu
 	var err error
 
 	// Pull applicable extension from list of processed extensions
-	ext, extensionIsRequired := r.Extensions[extensions.DockerExtensionKey]
+	dockerExt, dockerRequired, err := r.Extensions.GetDockerExtension()
+	if err != nil {
+		return nil, err
+	}
 
 	if args.AllowDockerHostAccess {
 		if driverName != DriverNameDocker {
 			return nil, errors.Errorf("allow-docker-host-access was enabled, but the driver is %s", driverName)
 		}
-		// Parse extension config to inform setup
-		config, ok := ext.(*extensions.Docker)
-		if !ok && extensionIsRequired {
-			return nil, errors.Errorf("unable to parse extension config: %+v", config)
-		}
-		driverImpl, err = r.dockerDriverWithHostAccess(config)
+
+		driverImpl, err = r.dockerDriverWithHostAccess(dockerExt)
 	} else {
-		if extensionIsRequired {
+		if dockerRequired {
 			return nil, errors.Errorf("extension %q is required but allow-docker-host-access was not enabled",
 				extensions.DockerExtensionKey)
 		}
@@ -63,7 +62,7 @@ func (r *Runtime) newDriver(driverName string, claimName string, args ActionArgu
 	return driverImpl, nil
 }
 
-func (r *Runtime) dockerDriverWithHostAccess(config *extensions.Docker) (driver.Driver, error) {
+func (r *Runtime) dockerDriverWithHostAccess(config extensions.Docker) (driver.Driver, error) {
 	const dockerSock = "/var/run/docker.sock"
 
 	if exists, _ := r.FileSystem.Exists(dockerSock); !exists {

--- a/pkg/porter/testdata/schema.json
+++ b/pkg/porter/testdata/schema.json
@@ -437,7 +437,10 @@
               "type": "object"
             }
           },
-          "type": "object"
+          "type": [
+            "string",
+            "object"
+          ]
         }
       ],
       "type": "array",

--- a/pkg/templates/templates/schema.json
+++ b/pkg/templates/templates/schema.json
@@ -238,7 +238,7 @@
       "type": "array",
       "items": [
         {
-          "type": "object",
+          "type": ["string","object"],
           "properties": {
             "docker": {
               "$schema": "http://json-schema.org/draft-07/schema#",

